pkg/executables/cmk.go

```diff
@@ -79,6 +79,39 @@ func (c *Cmk) ValidateTemplatePresent(ctx context.Context, profile string, domai
 	return nil
 }
 
+// SearchTemplatePresent - looks for template name or id and returns template name if available.
+func (c *Cmk) SearchTemplatePresent(ctx context.Context, profile string, template v1alpha1.CloudStackResourceIdentifier) (string, error) {
+	command := newCmkCommand("list templates")
+	applyCmkArgs(&command, appendArgs("templatefilter=all"), appendArgs("listall=true"))
+	if len(template.Id) > 0 {
+		applyCmkArgs(&command, withCloudStackId(template.Id))
+	} else {
+		applyCmkArgs(&command, withCloudStackName(template.Name))
+	}
+
+	result, err := c.exec(ctx, profile, command...)
+	if err != nil {
+		return "", fmt.Errorf("getting templates info - %s: %v", result.String(), err)
+	}
+	if result.Len() == 0 {
+		return "", fmt.Errorf("template %s not found", template)
```

Comment:
should this just return `"", nil`?
It's a good way of allowing the caller to distinguish between a not found result and an actual error.